models/news.model.js

```diff
@@ -26,17 +26,74 @@ exports.selectArticleById = (article_id) =>{
     return db
     .query(
         `SELECT * FROM articles
-        WHERE article_id = $1`,[article_id])
+        WHERE article_id = $1;`,[article_id])
     .then((result)=>{
         if(result.rowCount === 0){
-            return Promise.reject({ status:404, msg: 'Article Not Found'})
+            return Promise.reject({ status:404, msg: 'Not Found'})
         }else{
             return result.rows;
         }
         
     });
 }
 
+exports.selectCommentsByArticleId = (article_id) =>{
+    return db
+    .query(
+        `SELECT comment_id,body,votes,author,created_at FROM comments 
+        WHERE article_id = $1 
+        ORDER BY created_at desc;`, [article_id])
+        .then((result)=>{
+            if(result.rowCount !== 0){
+                return result.rows
+            }
+                return exports.selectArticleById(article_id)
+}).then((result)=>{
+    if(result.length === 1){
+        return []
+    }
+        return result;
+
+
+})
+}
+exports.addComment = (newComment, article_id) =>{
+    return exports.selectArticleById(article_id)
+    .then((result)=>{
+        if(result.length === 1){
+            const {username,body} = newComment
+            return db
+            .query(
+               `INSERT INTO comments
+                (author,body,article_id)
+                VALUES
+                ($1,$2,$3)
+                RETURNING *;`,[username,body,article_id])
+            .then((result)=>{
+                return  result.rows;
+            })
+        }
+    })
+}
+
+exports.updateVotesInArticles = (inc_votes,article_id)=>{
+    return exports.selectArticleById(article_id)
```

Comment:
Are you doing this to check if the article exists?
If so, no need! Just check the length of the rows array following the UPDATE

Reply:
yes, thats what i was trying to do. great, I'll work on this